shared/actions/fs/index.js

```diff
@@ -619,6 +619,13 @@ function* openPathItem(action: FsGen.OpenPathItemPayload): Saga.SagaGenerator<an
   )
 }
 
+const uploadFanOut = (state: TypedState, action: FsGen.UploadsPayload) =>
+  Saga.sequentially(
+    action.payload.localPaths.map(localPath =>
```

Comment:
why do this, why not just handle it natively instead of converting it ?

Reply:
You mean the existing `upload` handler should just take a `localPaths` array instead of just a single `localPath`?

Reply:
Each individual upload would be a separate RPC call though, so I figured doing a fan out here is cleaner.

Reply:
then why bother with this action at all, just emit the individual ones upstream

Reply:
I was trying to keep actionToPromise there (see the changes in `shared/actions/fs/platform-specific.desktop.js` below). If we need all these multiple actions, it seems like we'd need to use something like `takeEvery` there?